tools/tck-api/src/main/scala/org/opencypher/tools/tck/values/CypherValue.scala

```diff
@@ -104,9 +105,15 @@ object CypherPropertyMap {
 
 case class CypherPropertyMap(properties: Map[String, CypherValue] = Map.empty)
   extends CypherValue {
-  override def toString: String = s"{${properties.map {
-    case (k, v) => s"$k: $v"
-  }.mkString(", ")}}"
+
+  override def toString: String = toStringSorted
+
+  // Hack to make ordering work :/
+  private lazy val toStringSorted: String =
+    properties.toSeq
+      .sortBy { case (k, _) => k }
+      .map { case (k, v) => s"$k: $v" }
+      .mkString("{", ", ", "}")
```

Comment:
I agree this situation is not ideal and I too would prefer something more robust. This solves the bug because:

1. When comparing results we will compare unique row counts. The row count is a `Map[Map[String, CypherValue], Int]`, the row as key and count as value in map.
2. When creating that map we need to calculate the hash code for the rows. The hash code for CypherListValues is done on the sorted elements in the list. It is like this even for ordered lists, because they need to have the same hash code. The ordering in the list is (mostly) based on ordering `toString`.